app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/localBackend.ts

```diff
@@ -414,4 +412,24 @@ export class LocalBackend extends backend.Backend {
     override deleteTag() {
         return Promise.resolve()
     }
+
+    // === Protected methods ===
+
+    /** Get a fallback engine version. */
+    protected async getFallbackEngineVersion(): Promise<string> {
+        if (this.fallbackEngineVersion != null) {
```

Comment:
Shalln't `fallbackEngineVersion` be `Promise<string>` itself and just `resolve` itself to either `listVersions[0]` or some default value? This kind of _promise that reads a this field_ makes it hard to reason about the real flow of the code, in my opinion.

Reply:
1. I can read it
2. the state management is simplified
3. the `Future` is computed once and then returns always the same value
4. that wasn't clear to me from the previous version